control_plane/src/background_process.rs

```diff
@@ -126,16 +140,15 @@ where
                 thread::sleep(Duration::from_millis(RETRY_INTERVAL_MILLIS));
             }
             Err(e) => {
-                println!("{process_name} failed to start: {e:#}");
-                if let Err(e) = spawned_process.kill() {
-                    println!("Could not stop {process_name} subprocess: {e:#}")
-                };
```

Comment:
So this step was not taken in the "running out retries" case, and this is now moved to scopeguard with added waiting.

Reply:
Yes